pkg/cliplugins/workload/plugin/sync.go

```diff
@@ -161,34 +168,65 @@ func enableSyncerForWorkspace(ctx context.Context, config *rest.Config, workload
 	// Create a service account for the syncer with the necessary permissions. It will
 	// be owned by the workload cluster to ensure cleanup.
 	authResourceName := SyncerAuthResourcePrefix + workloadClusterName
-	sa, err := kubeClient.CoreV1().ServiceAccounts(namespace).Create(ctx, &corev1.ServiceAccount{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:            authResourceName,
-			OwnerReferences: workloadClusterOwnerReferences,
-		},
-	}, metav1.CreateOptions{})
-	if err != nil {
-		return "", fmt.Errorf("failed to create ServiceAccount: %w", err)
+	sa, err := kubeClient.CoreV1().ServiceAccounts(namespace).Get(ctx, authResourceName,
+		metav1.GetOptions{})
+
+	switch {
+	case errors.IsNotFound(err):
+		if sa, err = kubeClient.CoreV1().ServiceAccounts(namespace).Create(ctx, &corev1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:            authResourceName,
+				OwnerReferences: workloadClusterOwnerReferences,
+			},
+		}, metav1.CreateOptions{}); err != nil {
+			return "", fmt.Errorf("failed to create ServiceAccount: %w", err)
+		}
+	case err == nil:
+		//Overwrite the ownerref in case of a user changed it
+		sa.ObjectMeta.OwnerReferences = workloadClusterOwnerReferences
```

Comment:
It's fine to do sa.OwnerReferences (I saw you had a separate commit to adjust) - we usually omit the embedded ObjectMeta in the line

Reply:
It's potentially safer to add the ownerref if it doesn't exist, but don't overwrite any other existing ones (although I'm not sure how likely that case would be).